src/main/java/com/phonebookservice/controller/ContactController.java

```diff
@@ -0,0 +1,55 @@
+package main.java.com.phonebookservice.controller;
+
+import main.java.com.phonebookservice.exception.PhoneBookControllerException;
+import main.java.com.phonebookservice.model.Contact;
+import main.java.com.phonebookservice.model.Model;
+import main.java.com.phonebookservice.server.DBIf;
+import main.java.com.phonebookservice.util.Messages;
+import main.java.com.phonebookservice.util.Util;
+
+public class ContactController extends AbstractController {
+
+    /**
+     * contact controller constructor.
+     *
+     * @param databse the database
+     */
+    public ContactController(final DBIf databse) {
+        super(databse);
+    }
+
+    @Override
+    public final void create(final Model model) {
+        final DBIf database = getDatabase();
+
+        if (database == null) {
+            throw new PhoneBookControllerException(
+                    Messages.ERROR_DATABASE_NOT_FOUND);
+        }
+
+        if (model == null || !(model instanceof Contact)) {
+            throw new PhoneBookControllerException(
+                    Messages.ERROR_OBJECT_INVALID);
+        }
+
+        final Contact contact = (Contact) model;
+        final String errorMessage = validateContact(contact);
```

Comment:
why does this method  is an instance method ??!. It should be private static because it doesn't change in the controller state